src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/ViewComponentContext.cs

```diff
@@ -48,11 +48,6 @@ public ViewComponentContext()
                 throw new ArgumentNullException(nameof(viewComponentDescriptor));
             }
 
-            if (arguments == null)
```

Comment:
Arguments still can't be `null` can it? The `PropertyHelper.ObjectToDictionary` should always at least create a new dictionary for you.